File: test/Assets/Scripts/Rumble.cs
using UnityEngine;
using System.Collections;
using XInputDotNetPure;

public class Rumble : GameObjectSingleton<Rumble> {
	private float[] rumbling = new float[4];
	
	private IEnumerator StopVibration(PlayerIndex playerIndex, float time) {
		float currentTime = Time.time;
		rumbling[(int)playerIndex] = currentTime;
		yield return new WaitForSeconds(time);
	 	if (rumbling[(int)playerIndex] == currentTime) GamePad.SetVibration(playerIndex, 0.0f, 0.0f);
	}
	
	public void Vibrate(float big = 0.5f, float small = 0.5f, float time = 0.0f, int player = 1) {
		PlayerIndex  playerIndex = ToPlayerIndex(player);
		
	    GamePad.SetVibration (playerIndex, big, small);
	    if (time > 0.0f && (big > 0.0f || small > 0.0f)) StartCoroutine(StopVibration(playerIndex, time));
	}
	
	public void PadVibration (int playerIndex, float big, float small ) {
	    GamePad.SetVibration (ToPlayerIndex(playerIndex), big, small );
	}
	
	void OnApplicationQuit() {
		GamePad.SetVibration (PlayerIndex.One, 0,0);
		GamePad.SetVibration (PlayerIndex.Two, 0,0);
		GamePad.SetVibration (PlayerIndex.Three, 0,0);
		GamePad.SetVibration (PlayerIndex.Four, 0,0);
	}
	
	private PlayerIndex ToPlayerIndex(int playerIndex) {
		return PlayerIndex.One+(playerIndex-1);	
	}
}

File: test/Assets/Plugins/Rumble.js
#pragma strict

import XInputDotNetPure;
private var inverse : boolean;
private var time1 : float;
private var time2 : float; 

function Start () {
	if (PlayerPrefs.HasKey("inverse")) {
		inverse = (PlayerPrefs.GetInt("inverse") == -1 ? true : false);
	} else {
		PlayerPrefs.SetInt("inverse", 1);
		inverse = false;
	}
	
}

function Update () {
	if (Input.GetKeyDown(KeyCode.F12)) {
		inverse = !inverse;
		PlayerPrefs.SetInt("inverse", (inverse ? -1 : 1));
	} 
	if (time1 != Mathf.Infinity) {
		time1 -= Time.deltaTime;
		if (time1 <= 0.0f) {
			time1 = Mathf.Infinity;
			GamePad.SetVibration ( PlayerIndex.One,0.0f,0.0f);
			}
	}
	if (time2 != Mathf.Infinity) {
		time2 -= Time.deltaTime;
		if (time2 <= 0.0f) {
			time2 = Mathf.Infinity;
			GamePad.SetVibration ( PlayerIndex.Two,0.0f,0.0f);
		}
	}
}

function StopVibration(playerIndex : int, time : float) {
	yield WaitForSeconds(time);
 	Vibrate(playerIndex, 0.0f, 0.0f);
}


function Vibrate( playerIndex : int, val : float, time : float) {
	var player : PlayerIndex;
	var res : String = "";
	if (playerIndex == 1) {player = PlayerIndex.One;res += "one";}
	else if ( playerIndex == 2) {player = PlayerIndex.Two;res += "two";}
	else return;
	if (inverse) if (playerIndex == 1) player = PlayerIndex.Two;
	else player = PlayerIndex.One;
    GamePad.SetVibration ( player,val,val);
    if (time > 0.0f) {
    	if (player == PlayerIndex.One) time1 = Mathf.Infinity;
    	else time2 = Mathf.Infinity;
    	StartCoroutine(StopVibration(playerIndex, time));
    } else {
    	if (player == PlayerIndex.One) time1 = 0.2;
    	else time2 = 0.2;
    }
}